internal/witness/witness.go

```diff
@@ -196,16 +207,30 @@ func (w *witness) update(ctx context.Context, cp []byte, size uint64) ([]byte, e
 	body += "\n"
 	body += string(cp)
 
-	resp, err := w.post(ctx, w.url, body)
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, w.url, strings.NewReader(body))
+	if err != nil {
+		return nil, fmt.Errorf("failed to post to witness at %q: %v", w.url, err)
+	}
+	httpResp, err := w.client.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("failed to post to witness at %q: %v", w.url, err)
+	}
+	rb, err := io.ReadAll(httpResp.Body)
 	if err != nil {
 		return nil, fmt.Errorf("failed to post to witness at %q: %v", w.url, err)
 	}
```

Comment:
Worth disambiguating each of these error conditions in the msgs?

Reply:
You won't get a line number unless there's a `panic` - you'll only see the line number of the place where this (probably, by then, wrapped multiple times) error is ultimately passed into `log.Infof`.